### ISSUE TITLE
[SDK Refactor] Merge Viem changes into Passkeys branch

### ISSUE THREAD

**Issue by dasanra** [safe-global/safe-core-sdk#932]
## Context / issue

Viem migration is almost finished. As passkeys wasn't part of the main branch when the refactor started we should merge Viem changes into the passkeys branch and fix the existing conflicts so we can publish a version with both features in place.

## Proposed solution

 - Merge Viem branch into the Passkeys branch.
 - Once the Viem branch is finally merged into `development` we need to align the passkeys branch with `rebase --onto`.
 - Once the merge is ready we can test a preview of Viem + Passkeys in the test apps.
